From 196e704d9bd44a5bb7655d8a85b6f270b2165b3c Mon Sep 17 00:00:00 2001
From: Alfred Wingate <parona@protonmail.com>
Date: Mon, 20 Nov 2023 23:41:06 +0200
Subject: [PATCH] Only check for uic in qt5 if gui is enabled

We should only check for UIC (provided by qtwidgets) if we're building the GUI.

Qt (just qtcore) itself is *always* needed to build mkvtoolnix (it's used
for e.g. MIME type detection since 59.0.0), but the rest of Qt
(like qtwidgets) is only needed for the GUI build.

The build system incorrectly unconditionally checks for UIC. We fix it here
to only check for it when doing a GUI build.

https://bugs.gentoo.org/844097

Signed-off-by: Alfred Wingate <parona@protonmail.com>
--- a/ac/qt5.m4
+++ b/ac/qt5.m4
@@ -63,7 +63,7 @@ check_qt5() {
     AC_MSG_RESULT(no: could not find the rcc executable)
     return
 
-  elif test x"$UIC" = x; then
+  elif test x"$UIC" = x && test x"$enable_gui" = xyes; then
     AC_MSG_CHECKING(for Qt 5)
     AC_MSG_RESULT(no: could not find the uic executable)
     return
-- 
2.43.0

